pages/index.tsx

```diff
@@ -44,26 +42,19 @@ export default function HomePage({ packageCount }) {
 }
 
 export const getServerSideProps: GetServerSideProps = async (_context) => {
+  let packageCount;
   try {
-    const response = await fetch(`${API_URL}/api/count/packages/`);
-    //to prevent the case where site crashes due to error response
-    if (!response || response.status!==200) {
-      return {
-        props: {
-          packageCount: "all"
-        }
-      }
-    }
-    const { packageCount } = await response.json();
-    return {
-      props: {
-        packageCount
-      }
-    };
+    const response = await fetch(`${API_URL}/api/packages?limit=${Number.MAX_SAFE_INTEGER}`, {
+      method: 'HEAD'
+    });
+    packageCount = response.headers.get('x-total-count');
   } catch (error) {
-    // if package doesn't exist, return a 404
-    return {
-      notFound: true,
-    };
+    packageCount = null;
   }
+  return {
+    props: {
+      packageCount: packageCount || ''
+    }
+  };
```

Comment:
You don't need a default value here since packageCount gets initialized either through the response or with null in the catch

Reply:
Also for info, if you set a default ``, then packageCount type becomes `number | string`so better not do that 😅